internal/provider/mcc_schema_1_4.go

```diff
@@ -617,13 +664,22 @@ type launchpadSchema14ModelSpecMCR struct {
 }
 
 type launchpadSchema14ModelSpecMKE struct {
-	AdminPassword   types.String `tfsdk:"admin_password"`
-	AdminUsername   types.String `tfsdk:"admin_username"`
-	ImageRepo       types.String `tfsdk:"image_repo"`
-	Version         types.String `tfsdk:"version"`
-	InstallFlags    types.List   `tfsdk:"install_flags"`
-	UpgradeFlags    types.List   `tfsdk:"upgrade_flags"`
-	LicenseFilePath types.String `tfsdk:"license_file_path"`
+	AdminPassword   types.String                         `tfsdk:"admin_password"`
+	AdminUsername   types.String                         `tfsdk:"admin_username"`
+	ImageRepo       types.String                         `tfsdk:"image_repo"`
+	Version         types.String                         `tfsdk:"version"`
+	InstallFlags    types.List                           `tfsdk:"install_flags"`
+	UpgradeFlags    types.List                           `tfsdk:"upgrade_flags"`
+	LicenseFilePath types.String                         `tfsdk:"license_file_path"`
+	ConfigFile      types.String                         `tfsdk:"config_file"`
+	ConfigData      types.String                         `tfsdk:"config_data"`
+	Cloud           []launchpadSchema14ModelSpecMKECloud `tfsdk:"cloud"`
```

Comment:
our current approach for optional blocks, such as this cloud block, is to use a nested block, with a size limit validatoin.  To match with this we must us a list for unmarshalling, even though we only ever have 0|1 element.